src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/ResponseEventsAboutToSubmit.java

```diff
@@ -82,17 +87,75 @@ private void validateChangedCaseCode(PreSubmitCallbackResponse<SscsCaseData> res
         final Optional<CaseDetails<SscsCaseData>> caseDetailsBefore = callback.getCaseDetailsBefore();
 
         if (response.getData().getAppeal().getBenefitType() != null
-            && Benefit.CHILD_SUPPORT.getShortName().equalsIgnoreCase(response.getData().getAppeal().getBenefitType().getCode())
-            && !Benefit.CHILD_SUPPORT.getCaseLoaderKeyId().contains(response.getData().getBenefitCode())
             && caseDetailsBefore.isPresent()
             && caseDetailsBefore.get().getCaseData().getBenefitCode() != null
-            && !caseDetailsBefore.get().getCaseData().getBenefitCode().equals(callback.getCaseDetails().getCaseData().getBenefitCode())) {
+            && !caseDetailsBefore.get().getCaseData().getBenefitCode()
+            .equals(callback.getCaseDetails().getCaseData().getBenefitCode())) {
 
-            if (response.getData().getOtherParties() != null && response.getData().getOtherParties().size() > 0) {
-                response.addError("Benefit code cannot be changed on cases with registered 'Other Party'");
-            } else {
-                response.addWarning("The benefit code will be changed to a non-child support benefit code");
+            if (Benefit.CHILD_SUPPORT.getShortName()
+                .equalsIgnoreCase(response.getData().getAppeal().getBenefitType().getCode())
+                && !Benefit.CHILD_SUPPORT.getCaseLoaderKeyId().contains(response.getData().getBenefitCode())) {
+                if (response.getData().getOtherParties() != null && response.getData().getOtherParties().size() > 0) {
+                    response.addError("Benefit code cannot be changed on cases with registered 'Other Party'");
+                } else {
+                    response.addWarning("The benefit code will be changed to a non-child support benefit code");
+                }
+            }
+
+            if ((isSscs5Case(response.getData().getAppeal().getBenefitType().getCode())
+                && !buildSscs5BenefitCaseLoaderKeyId().contains(response.getData().getBenefitCode()))
+                || (!isSscs5Case(response.getData().getAppeal().getBenefitType().getCode())
+                && buildSscs5BenefitCaseLoaderKeyId().contains(response.getData().getBenefitCode()))) {
+                response.addError("Benefit code cannot be changed to the selected code");
+            }
+        }
+    }
+
+    private boolean isSscs5Case(String benefitName) {
+        if(benefitName != null) {
+            Optional<Benefit> benefitOptional = Benefit.findBenefitByShortName(benefitName);
+            if (benefitOptional.isPresent()) {
```

Comment:
could use filter instead of these 3 lines

Reply:
Put a filter